### PR TITLE
Adds storage buffer bindings

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
This PR adds storage buffer binding type. These are similar to uniform buffers but can be larger and allow for runtime sized arrays and can be used to retrive data from the gpu.

p.s. These could be used to remove the maximum number of lights in the pbr shader by binding as storage buffer with a runtime sized array.

p.p.s Next thing to tackle is the computer shader.

### CODE REVIEW

**Review by Lowentwickler — APPROVED** [review]
Thank you for the PR! Good work!

**Comment by Lowentwickler** [comment]
What can be a good and simple example demonstrating the `Storage` buffer in action?

**Comment by Lowentwickler** [comment]
Could you please rebase your branch so I can merge it?

File: dotrix_core/src/renderer.rs [hunk 1]
```diff
@@ -308,6 +298,8 @@ pub enum Binding<'a> {
     Texture3D(&'a str, Stage, &'a TextureBuffer),
     /// Texture sampler binding
     Sampler(&'a str, Stage, &'a Sampler),
+    /// Texture sampler binding
```
[the remark this answers is not in the document]
Comment (Lowentwickler) [reply dotrix_core/src/renderer.rs:301]: yep

**Comment by QuantumEntangledAndy** [comment]
> Could you please rebase your branch so I can merge it?

I cannot see any merge conflicts did I base it onto the wrong branch or something? Sorry for not understanding

**Comment by Lowentwickler** [comment]
It worked with a squash. No issue :)

**Comment by QuantumEntangledAndy** [comment]
Oh I just fixed that doc comment a bit late... :(

**Comment by Lowentwickler** [comment]
Sorry, my fault. If you submit a new PR with this little fix I will merge it

**Pull request merged**